src/app/applications/time_lock/page.mdx

```diff
@@ -0,0 +1,372 @@
+        address owner;
+        mapping(bytes32 => bool) queued;
+    }
+}
+
+// Error types for the TimeLock contract
+#[derive(SolidityError)]
+pub enum TimeLockError {
+    // Error for when the contract is already initialized.
+    AlreadyInitialized(AlreadyInitialized),
+    // Error for when the sender is not the owner
+    NotOwnerError(NotOwnerError),
+    // Error for when the transaction is already queued
+    AlreadyQueuedError(AlreadyQueuedError),
+    // Error for when the timestamp is not in the range
+    TimestampNotInRangeError(TimestampNotInRangeError),
+    // Error for when the transaction is not queued
+    NotQueuedError(NotQueuedError),
+    // Error for when the timestamp has not yet passed
+    TimestampNotPassedError(TimestampNotPassedError),
+    // Error for when the timestamp has expired
+    TimestampExpiredError(TimestampExpiredError),
+    // Error for when a transaction fails
+    TxFailedError(TxFailedError),
+}
+
+// Marks `TimeLock` as a contract with the specified external methods
+#[external]
+impl TimeLock  {
+
+    // Minimum delay allowed for a transaction
+    pub const MIN_DELAY: u64 = 10;
+    // Maximum delay allowed for a transaction
+    pub const MAX_DELAY: u64 = 1000;
+    // Grace period after the maximum delay
+    pub const GRACE_PERIOD: u64 = 1000;
+
+    pub fn initialize(&mut self) -> Result<(), TimeLockError> {
+        if self.owner.get() != Address::default() {
+            return Err(TimeLockError::AlreadyInitialized(AlreadyInitialized{}))
+        }
+        self.owner.set(msg::sender());
+        Ok(())
+    }
+
+    pub fn owner(&self) -> Address {
+        self.owner.get()
+    }
+
+    // Function to generate a transaction ID
+    pub fn get_tx_id(
+        &self, 
+        target: Address, // Target address for the transaction
+        value: U256, // Value to be transferred
+        func: String, // Function name to be called
+        data: Bytes, // Data to be passed to the function
+        timestamp: U256, // Timestamp for the transaction
+    ) -> FixedBytes<32>{
+        
+        // Package the transaction data
+        let tx_hash_data = (target, value, func, data, timestamp);
+        // Encode the transaction data using ABI encoding
+        let tx_hash_bytes = TxIdHashType::abi_encode_sequence(&tx_hash_data);
+        // Initialize a new Keccak256 hasher
+        let mut hasher = Keccak256::new();
+        // Update the hasher with the encoded bytes
+        hasher.update(tx_hash_bytes);
+        // Finalize the hash computation
+        let result = hasher.finalize();
+        // Convert the hash result to a vector
+        let result_vec = result.to_vec();
+        // Create a FixedBytes<32> instance from the result vector
+        // This is used as the transaction ID
+        alloy_primitives::FixedBytes::<32>::from_slice(&result_vec)
+    }
+
+    // The `deposit` method is payable, so it can receive funds.
+    #[payable]
+    pub fn deposit(&self) {
```

Comment:
What's the purpose of this function?

Reply:
Because we need to deposit ethers into this contract if the tx needs to send ethers, in solidity by example they just use `receive` method but we don't support that so I made `deposit` here

Reply:
Oh, I totally missed the receive() method in the Solidity counterpart. Thanks!